relay-replays/tests/fixtures/rrweb-pii.json

```diff
@@ -0,0 +1,33 @@
+[
+    {
+        "type": 3,
+        "data": {
+            "source": 0,
+            "texts": [],
+            "attributes": [],
+            "removes": [],
+            "adds": [
+                {
+                    "parentId": 74,
+                    "nextId": null,
+                    "node": {
+                        "type": 2,
+                        "tagName": "h1",
+                        "attributes": {
+                            "id": "react-tooltip"
+                        },
+                        "childNodes": [
+                            {
+                                "type": 3,
+                                "textContent": "foo",
+                                "id": 284
+                            }
+                        ],
+                        "id": 283
+                    }
+                }
+            ]
+        },
+        "timestamp": 1665063926945
+    }
+]
```

Comment:
Is this fixture supposed to contain PII? I noticed there is no test that actually loads this file.

Reply:
Good spot.  I added coverage but it doesn't seem to be stripping the PII.  This condition never returns true: https://github.com/getsentry/relay/blob/master/relay-general/src/pii/processor.rs#L42

Any idea what needs to be done?

Reply:
Seems that the following condition was the problem, i.e. the matcher never matched because we're plugging in a root element:

https://github.com/getsentry/relay/blob/68526d4fe86e5597250124d64b1fef1174077948/relay-general/src/processor/attrs.rs#L598

I checked out this branch and added 

```rust
let processing_state = processing_state.enter_borrowed(
      "",
      processing_state.inner_attrs(),
      Some(ValueType::String),
);
```

This makes the unit test pass, but it feels quite hacky, maybe @jan-auer has a better idea? At this point it might actually be easier to use `Annotated`.

Reply:
@jan-auer ping